### ISSUE TITLE
Long running job in AWS Kubernetes

### ISSUE THREAD

**Issue by vijtad** [rq/rq#1516]
There is a long running job in Kubernetes performing Horizontal Pod Auto scaling based on queue length.
It is issuing SIGTERM and our worker is killed in the middle.
Is SIGTERM handled gracefully by rq worker ?

Also, I would like to know the command to  get the worker id number of currently  running process.

**Comment by rpkak** [comment]
> There is a long running job in Kubernetes performing Horizontal Pod Auto scaling based on queue length.
> It is issuing SIGTERM and our worker is killed in the middle.
> Is SIGTERM handled gracefully by rq worker ?

If no job is running on the worker, SIGTERM will gracefully stop the worker. Else the worker will be stopped after the job ends (Warm shutdown). If you send another SIGTERM to the worker, the worker will terminate the process of the the job (I think with SIGKILL).

If you can't work with this behavior you could look at `HerokuWorker`:

https://github.com/rq/rq/blob/77e7ef6983871175a51bd10f340c6020765e2a26/rq/worker.py#L1187-L1193

Or you can just build your own worker (and make a PR so everyone can use it).

You start a worker which is not the `Worker` class using:

```
rq worker -w <workerclass>
```

E.g.:

```
rq worker -w rq.worker.HerokuWorker
```

> Also, I would like to know the command to get the worker id number of currently running process.

I'm not sure if you want to do this in python (from inside of a job) or sh. Here are both possibilities:

__Python:__

Just execute:

```python
from rq import get_current_job
get_current_job().worker_name
```

__sh, bash, etc:__

```
rq info
```

returns you a general overview about the rq workers and queues.


```
rq info -W
```

only returns the workers.

And

```
rq info -Wr
```

Makes the returned string easier to parse.

**Comment by vijtad** [comment]
Thank you for prompt answer.

In Kubernetes, Horizontal Pod Auto Scaling yaml file - I used RQ exporter to get scale the number of workers.
In my deployment yaml file, I set the graceperiod to 2 hours as i am running long running jobs
I set it same as 2 hours in my enque job call timeout parameter.
The original worker worked very well with SIGTERM invoked by Kubernetes.
After grace period, Kubernetes will send SIGKILL to kill the pod. But the  job normally before in time which will shutdown graciously.
